codegen/src/test/resources/software/amazon/awssdk/codegen/poet/rules/endpoint-provider-class.java

```diff
@@ -57,6 +53,316 @@ private static Map<Identifier, Value> toIdentifierValueMap(QueryEndpointParamete
         if (endpointId != null) {
             paramsMap.put(Identifier.of("endpointId"), Value.fromStr(endpointId));
         }
+        Boolean defaultTrueParam = params.defaultTrueParam();
+        if (defaultTrueParam != null) {
+            paramsMap.put(Identifier.of("defaultTrueParam"), Value.fromBool(defaultTrueParam));
+        }
+        String defaultStringParam = params.defaultStringParam();
+        if (defaultStringParam != null) {
+            paramsMap.put(Identifier.of("defaultStringParam"), Value.fromStr(defaultStringParam));
+        }
+        String deprecatedParam = params.deprecatedParam();
+        if (deprecatedParam != null) {
+            paramsMap.put(Identifier.of("deprecatedParam"), Value.fromStr(deprecatedParam));
+        }
         return paramsMap;
     }
+
+    private static EndpointRuleset ruleSet() {
```

Comment:
How performant is executing these rule sets?

Reply:
Do you mean building them here?

Reply:
If you're asking about evaluating the ruleset against a given set of params, I'm curious myself. I'll get some numbers tomorrow!

Just building `EndpointRuleSet` should be pretty fast for the most part. It's just creating some POJO's, and (non-regex based) string parsing

Reply:
Yep, I meant executing them.

Reply:
Yep makes sense I'll get the numbers later today when I get some time